### ISSUE TITLE
Selecting a nonexistent field using record dot yields error about missing HasField instance

### ISSUE THREAD

**Issue by MorrowM** [haskell/error-messages#539]
Using GHC 9.4.4:
```
λ> :set -XOverloadedRecordDot
λ> data Person = Person { name :: String, age :: Int } deriving (Eq, Show, Ord)
λ> john = Person "John" 34
λ> john.address

<interactive>:4:1: error:
    • No instance for (GHC.Records.HasField "address" Person a0)
        arising from selecting the field ‘address’
    • In the expression: john.address
      In an equation for ‘it’: it = john.address
```

While this is technically a "correct" error message I believe we can do better. For example

```
<interactive>:4:1: error:
    • Cannot select nonexistent field ‘address’
        from a record of type ‘Person’
    • In the expression: john.address
      In an equation for ‘it’: it = john.address
```
Whether to mention `HasField` at all is perhaps a question, but I believe that if a user is advanced enough to potentially want to mess with `HasField` then they don't need the error message to spell that out for them.

**Comment by dixonary** [comment]
Seconding this. It's severe enough that I really think it will slow down the take-up of record dot syntax, especially among new learners.

Anything of the general shape that Morrow suggests would be streets ahead.

**Comment by Kleidukos** [comment]
Agreed, we can score big time with this feature amongst new users and this needs to be a prime concern.

**Comment by MorrowM** [comment]
The relevant GHC issue:
https://gitlab.haskell.org/ghc/ghc/-/issues/22382

**Comment by massung** [comment]
I know this issue is old, but wanted to add that - as a hobbyist Haskell user - this error popped up and it took me _way_ too long to figure out that the actual issue was I was missing an import; the record wasn't in scope.

```
No instance for ‘GHC.Records.HasField
                         "login" Persist.User.User String’
        arising from selecting the field ‘login’
```

Having the error message be as suggested in the issue - and perhaps adding a little hint that the record type may not be in scope (not imported or misspelled, as with other error messages), would be very helpful.

**Comment by unhammer** [comment]
Also, HLS says `typecheck [GHC-39999]: • Could not deduce ‘HasField "login" User a0’` but https://errors.haskell.org/messages/GHC-39999/ is not very helpful here.

The error message now includes a line `NB: There is no field selector 'login :: User -> a0' in scope for record type 'User'` but I need to scroll a bit to find it below a bunch of noise about bindings and contexts, it should be the *first* line of the error message.
